packages/next/src/shared/lib/get-img-props.ts

```diff
@@ -465,6 +465,18 @@ export function getImgProps(
             `Image with src "${src}" has invalid "height" property. Expected a numeric value in pixels but received "${height}".`
           )
         }
+        // eslint-disable-next-line no-control-regex
+        if (/^[\x00-\x20]/.test(src)) {
+          throw new Error(
+            `Image with src "${src}" cannot start with a space or control character. Use src.trimStart() to remove it or encodeURIComponent(src) to keep it.`
+          )
+        }
+        // eslint-disable-next-line no-control-regex
+        if (/[\x00-\x20]$/.test(src)) {
+          throw new Error(
+            `Image with src "${src}" cannot end with a space or control character. Use src.trimEnd() to remove it or encodeURIComponent(src) to keep it.`
```

Comment:
I don't quite follow. Is the PR description referring to [this line](https://github.com/vercel/next.js/blob/c1f8d9317588e51a8a31240f6add36b5f2c9f9bf/packages/next/src/shared/lib/image-loader.ts#L35)? If yes, then I don't see how users can chose to preserve the leading or trailing whitespace. This code branch is only hit for (potentially) absolute URLs, right? But encoding `src` with `encodeURIComponent` would yield a string that is invalid to be passed into the `URL` constructor:

```js
encodeURIComponent('https://example.com/foo ')
// 'https%3A%2F%2Fexample.com%2Ffoo%20'
```

Reply:
Ah ok, I guess this is not the line is question, since this is part of the default loader, and you're talking about the handling of the URL in the API. Then ignore my comment if it does not apply.

Reply:
The line of code where the error is thrown is for any `src` string, not just absolute urls.

The PR is really meant for users who do not intend to run `next start`, but we don't know that during `next dev` so its best to fail fast.